compiler/next/include/chpl/uast/UseClause.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_USECLAUSE_H
+#define CHPL_UAST_USECLAUSE_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Decl.h"
+#include "chpl/uast/Expression.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents an individual use clause. For example:
+
+  \rst
+  .. code-block:: chapel
+    // This contains two use clauses, 'Foo as X' and 'Baz as Y'.
+    use Foo as X, Baz as Y;
+
+  \endrst
+ */
+class UseClause final : public Expression {
+ public:
+  enum LimitationClauseKind {
```

Comment:
`enum class` or `enum struct` is preferred. What do these values represent?

Reply:
I think the current strategy is to use old enums for things that are already namespaced by a class. The reason why I got away with using `enum class` for `BlockStyle` is because it lives outside of any one node and the values needed to be namespaced by _something_.

Reply:
I'll add a comment, thanks!

Reply:
Yeah. A reasonable alternative strategy would be to put all of these enums in a header file for the enums (rather than defining them within the class). Anyway it seems to me that `UseClause::LimitationClauseKind::EXCEPT` is a bit much for a long name so I wouldn't want to see both of those required.